ui/v2.5/src/components/ScenePlayer/autostart-button.ts

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import videojs, { VideoJsPlayer } from "video.js";
+
+interface IAutostartButtonOptions {
+  enabled?: boolean;
+}
+
+interface AutostartButtonOptions extends videojs.ComponentOptions {
+  autostartEnabled: boolean;
+}
+
+class AutostartButton extends videojs.getComponent("Button") {
+  private autostartEnabled: boolean;
+
+  constructor(player: VideoJsPlayer, options: AutostartButtonOptions) {
+    super(player, options);
+    this.autostartEnabled = options.autostartEnabled;
+    this.updateIcon();
+  }
+
+  buildCSSClass() {
+    return `vjs-autostart-button ${super.buildCSSClass()}`;
+  }
+
+  private updateIcon() {
+    this.removeClass("vjs-icon-play-circle");
+    this.removeClass("vjs-icon-cancel");
+
+    if (this.autostartEnabled) {
+      this.addClass("vjs-icon-play-circle");
+      this.controlText(this.localize("Auto-start enabled (click to disable)"));
```

Comment:
Could we accept localised strings in the constructor options instead? That way we could put the strings into the localisation files.